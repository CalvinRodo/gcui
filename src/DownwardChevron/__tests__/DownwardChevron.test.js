import React from 'react'
import { mount, shallow } from 'enzyme'
import 'jest-styled-components'
import toJSON from 'enzyme-to-json'
import DownwardChevron from '../../DownwardChevron'

describe('<DownwardChevron />', () => {
  it('can be rendered', () => {
    let wrapper = shallow(<DownwardChevron />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('accepts a width prop', () => {
    let wrapper = shallow(<DownwardChevron width="1000%" />).find('svg')
    expect(wrapper.props().width).toEqual('1000%')
  })

  it('defaults to width of 0.7em', () => {
    let wrapper = shallow(<DownwardChevron />).find('svg')
    expect(wrapper.props().width).toEqual('0.7em')
  })

  it('accepts a colour', () => {
    let wrapper = shallow(<DownwardChevron colour="#abc" />)
    let path = wrapper.find('path')
    expect(path.props().style.fill).toEqual('#abc')
  })

  it('allows the padding around the chevron to be set', () => {
    let wrapper = shallow(
      <DownwardChevron verticalPadding="10vh" horizontalPadding="4em" />,
    ).dive()
    expect(toJSON(wrapper)).toHaveStyleRule('padding', '10vh 4em')
  })

  it('the padding has default values of 0.3em', () => {
    let wrapper = shallow(<DownwardChevron />).dive()
    expect(toJSON(wrapper)).toHaveStyleRule('padding', '0.3em 0.3em')
  })
})
