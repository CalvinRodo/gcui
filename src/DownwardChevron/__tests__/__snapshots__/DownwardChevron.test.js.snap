// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownwardChevron /> can be rendered 1`] = `
<styled.span
  horizontal="0.3em"
  vertical="0.3em"
>
  <svg
    aria-label="downward chevron"
    height="100%"
    preserveAspectRatio="xMinYMin meet"
    version="1.1"
    viewBox="0 0 5.1385611 3.2992621"
    width="0.7em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      transform="matrix(0.39946474,0,0,0.39946474,0.53101811,2.4482309)"
    >
      <path
        d="M 11.382674,-3.9343035 5.4615799,1.9788107 q -0.151619,0.1516183 -0.359096,0.1516183 -0.207478,0 -0.359096,-0.1516183 l -5.921094,-5.9131142 q -0.151618,-0.151618 -0.151618,-0.359096 0,-0.215457 0.151618,-0.367076 l 1.32466503,-1.316685 q 0.151618,-0.151618 0.359096,-0.151618 0.207478,0 0.359096,0.151618 l 4.23733297,4.237333 4.2373325,-4.237333 q 0.151618,-0.151618 0.359096,-0.151618 0.2074778,0 0.3590966,0.151618 l 1.324665,1.316685 q 0.151618,0.151619 0.151618,0.367076 0,0.207478 -0.151618,0.359096 z"
        style={
          Object {
            "fill": "#FFF",
          }
        }
      />
    </g>
  </svg>
</styled.span>
`;
